### PR TITLE
FIO Job File support

### PR DESCRIPTION
**Pull request by saibulusu**
Adding Job File support to the FIOExecutor class. Created a profile PERF-IO-FIO-OLTP to expose this feature. Tested on linux-x64, linux-arm64, windows-x64 VMs.

### CODE REVIEW

**Comment by yangpanMS** [comment]
Could we include unit test for the fio job file parsing?

File: website/docs/guides/0011-profiles.md
Comment (yangpanMS) [inline website/docs/guides/0011-profiles.md:1]: Please add document specific to FIO

File: src/VirtualClient/VirtualClient.Actions/FIO/FioDiscoveryExecutor.cs [hunk 1]
```diff
@@ -52,6 +52,23 @@ public List<string> BlockSize
             }
         }
 
+        /// <summary>
+        /// The size of the test file that should use in workload tests (e.g. 496GB).
```
Comment (yangpanMS) [inline src/VirtualClient/VirtualClient.Actions/FIO/FioDiscoveryExecutor.cs:56]: Is this used anywhere?

**Comment by saibulusu** [comment]
> Could we include unit test for the fio job file parsing?

I have added some unit tests now. None of the unit tests referenced the architecture except for the one I added, so I set the architecture to X64.

File: src/VirtualClient/VirtualClient.Actions/FIO/FioDiscoveryExecutor.cs (hunk 1 above)
Comment (saibulusu) [reply src/VirtualClient/VirtualClient.Actions/FIO/FioDiscoveryExecutor.cs:56]: FileSize was inherited from DiskWorkloadExecutor (via FioExecutor) and used within FioDiscoveryExecutor. I removed the dependency between FioExecutor and DiskWorkloadExecutor, so I had to add this field here.
It is used in the ExecuteAsync method in DiskWorkloadExecutor.

**Comment by saibulusu** [comment]
The FioExecutor currently appends --ioengine & --filename to the command for the command line approach. I think it would make sense to keep the same in the job file approach, so I think we should expect the template job file that is provided to exclude the ioengine & file path parameters.

File: website/docs/guides/0011-profiles.md
Comment (saibulusu) [reply website/docs/guides/0011-profiles.md:1]: Added a profile section for PERF-IO-FIO-OLTP.json.

**Review by brdeyo — CHANGES_REQUESTED** [review]

File: website/docs/workloads/fio/fio-profiles.md [hunk 2]
```diff
@@ -426,4 +426,89 @@ This profile uses an algorithm to determine the amount of IOPS to run against th
...
+Runs an IO-intensive workload using the Flexible IO Tester (FIO) toolset. Multi-throughput OLTP-C workload to emulate a SQL Server OLTP disk 
+workload by running four workload compononents in-parallel: random reads, random writes, sequential reads and sequential writes each with an overall 
+weight/percentage defined. 
+A weight of 0 for and of the workload components will cause that component to be excluded from the overall operations. 
+
+Random IO : It represents the Database of OLTP-C workload.
+Sequential IO : It represents the logs of OLTP-C workload.
+Therefore, they are performed on different disks
+
+The intent of this profile is to expose the ability to use template job files via the FioExecutor.
```
[the remark this answers is not in the document]
Comment (saibulusu) [reply website/docs/workloads/fio/fio-profiles.md]: Removed this sentence.

File: website/docs/guides/0011-profiles.md [hunk 3]
```diff
@@ -68,6 +68,7 @@ the system. The following table describes the set of well-known parameters that
 | SystemMemoryKilobytes                 | Represents the total memory/RAM (in kilobytes) on the system. Note that industry standard memory unit definitions are used (e.g. 1 kilobyte = 1024 bytes). |
 | SystemMemoryMegabytes                 | Represents the total memory/RAM (in megabytes) on the system. Note that industry standard memory unit definitions are used (e.g. 1 megabyte = 1024 kilobytes or 1024 x 1024 bytes). |
 | SystemMemoryGigabytes                 | Represents the total memory/RAM (in gigabytes) on the system. Note that industry standard memory unit definitions are used (e.g. 1 gigabyte = 1024 megabytes or 1024 x 1024 kilobytes or 1024 x 1024 x 1024 bytes). |
+| ScriptPath:\{script_folder\}			| Represents the path to a script that is copied to the build output 'scripts' directory. |
```
[the remark this answers is not in the document]
Comment (saibulusu) [reply website/docs/guides/0011-profiles.md]: Changed to alphabetical order.

File: src/VirtualClient/VirtualClient.Main/profiles/PERF-IO-FIO-OLTP.json [hunk 4]
```diff
@@ -0,0 +1,282 @@
...
+      "Type": "FioExecutor",
+      "Parameters": {
+        "Scenario": "FioExecutorJobFile_4K_BlockSize",
+        "PackageName": "fio",
+        "ProcessModel": "$.Parameters.ProcessModel",
+        "TestName": "read_write_operations_4k_blocksize",
+        "JobFiles": "{ScriptPath:fio}/oltp-c2.fio.jobfile",
+        "FileName": "fio-test.dat",
+        "DeleteTestFilesOnFinish": false,
+        "DiskFilter": "$.Parameters.DiskFilter",
```
[the remark this answers is not in the document]
Comment (saibulusu) [reply src/VirtualClient/VirtualClient.Main/profiles/PERF-IO-FIO-OLTP.json:42]: Removed these parameters, some are directly set in the template job file and others are not needed at all.

File: src/VirtualClient/VirtualClient.Main/profiles/PERF-IO-FIO-OLTP.json [hunk 5]
```diff
@@ -0,0 +1,282 @@
...
+  "Metadata": {
+    "RecommendedMinimumExecutionTime": "00:20:00",
+    "SupportedPlatforms": "linux-x64,linux-arm64,win-x64",
+    "SupportedOperatingSystems": "CBL-Mariner,CentOS,Debian,RedHat,Suse,Ubuntu,Windows",
+    "Notes": "This profile uses FIO job files (e.g. *.jobfile). The job file paths should be comma/semicolon-delimited under 'JobFiles'.",
+    "Documentation": "FIO documentation: https://fio.readthedocs.io/en/latest/fio_doc.html FIO Job File documentation: https://fio.readthedocs.io/en/latest/fio_doc.html#job-file-format"
+  },
+  "Parameters": {
+    "DiskFillSize": "10G",
+    "FileSize": "10G",
```
[the remark this answers is not in the document]
Comment (saibulusu) [reply src/VirtualClient/VirtualClient.Main/profiles/PERF-IO-FIO-OLTP.json]: I set it to 124 GB and 20 GB to match the multithroughput profile, which runs OLTP as well and has a very similar template job file.

File: website/docs/workloads/fio/fio-profiles.md [hunk 6]
```diff
@@ -426,4 +426,89 @@ This profile uses an algorithm to determine the amount of IOPS to run against th
...
   ./VirtualClient --profile=PERF-IO-FIO-MULTITHROUGHPUT.json --system=Demo --timeout=1440 --packageStore="{BlobConnectionString|SAS Uri}"  --parameters="TargetPercents="40,80,120"
+  ```
+
+
+-----------------------------------------------------------------------
+
+## PERF-IO-FIO-OLTP.json
+Runs an IO-intensive workload using the Flexible IO Tester (FIO) toolset. Multi-throughput OLTP-C workload to emulate a SQL Server OLTP disk 
+workload by running four workload compononents in-parallel: random reads, random writes, sequential reads and sequential writes each with an overall 
+weight/percentage defined.
```
[the remark this answers is not in the document]
Comment (saibulusu) [reply website/docs/workloads/fio/fio-profiles.md]: Removed the word "defined" and showed the example for the values that are used in this profile.

File: website/docs/workloads/fio/fio-profiles.md [hunk 7]
```diff
@@ -426,4 +426,89 @@ This profile uses an algorithm to determine the amount of IOPS to run against th
...
+  | InitializeDisksInParallel | Optional. Specifies whether uninitialized/unformatted disks on the system should be initialized + formatted in parallel. | true (initialized in-parallel) |
+  | SequentialDiskCount | Optional. Specifies the number of disk that will have Sequential I/O from Selected Disks. | 1 |
+  
+  
+* **Profile Component Parameters** 
+  The following section describes the parameters used by the individual components in the profile.
+
+  | Parameter                 | Purpose                                                                         | 
+  |---------------------------|---------------------------------------------------------------------------------|
+  | DefaultRandomIOBlockSize  | Default Block size value for Random Read and Write. |
```
[the remark this answers is not in the document]
Comment (saibulusu) [reply website/docs/workloads/fio/fio-profiles.md]: Removed those parameters.

**Review by brdeyo — APPROVED** [review]

**Pull request merged**